app/src/main/java/bg/zahov/app/ui/settings/SettingsScreen.kt

```diff
@@ -0,0 +1,282 @@
+        },
+        theme = uiState.data.theme,
+        changeTheme = { onDismiss ->
+            RadioSettingsDialog(
+                title = THEME_SETTING,
+                items = theme,
+                onDismissRequest = onDismiss,
+                selected = uiState.data.theme,
+                setSelected = { viewModel.writeNewSetting(title = THEME_SETTING, it) })
+
+        },
+        units = uiState.data.units,
+        changeUnits = { onDismiss ->
+            RadioSettingsDialog(
+                title = UNIT_SETTING,
+                items = units,
+                onDismissRequest = onDismiss,
+                selected = uiState.data.units,
+                setSelected = { viewModel.writeNewSetting(title = UNIT_SETTING, it) })
+
+        },
+        language = uiState.data.language,
+        changeLanguage = { onDismiss ->
+            RadioSettingsDialog(
+                title = LANGUAGE_SETTING,
+                items = languages,
+                onDismissRequest = onDismiss,
+                selected = uiState.data.language,
+                setSelected = { viewModel.writeNewSetting(title = LANGUAGE_SETTING, it) })
+
+        },
+        navigateEditProfile = { navigateEditProfile() },
+        enableSound = uiState.data.soundEffects,
+        enableSoundChange = {
+            viewModel.writeNewSetting(
+                title = SOUND_EFFECTS_SETTING,
+                newValue = !uiState.data.soundEffects
+            )
+        },
+        enableSync = uiState.data.automaticSync,
+        enableSyncChange = {
+            viewModel.writeNewSetting(
+                title = AUTOMATIC_SYNC_SETTING,
+                newValue = !uiState.data.automaticSync
+            )
+        },
+        enableVibrate = uiState.data.vibration,
+        enableVibrateChange = {
+            viewModel.writeNewSetting(
+                title = VIBRATION_SETTING,
+                newValue = !uiState.data.vibration
+            )
+        },
+        showUpdate = uiState.data.updateTemplate,
+        showUpdateChange = {
+            viewModel.writeNewSetting(
+                title = UPDATE_TEMPLATE_SETTING,
+                newValue = !uiState.data.updateTemplate
+            )
+        },
+        enableWatch = uiState.data.enableWatch,
+        useWatchChange = {
+            viewModel.writeNewSetting(
+                title = WATCH_SETTINGS,
+                newValue = !uiState.data.enableWatch
+            )
+        },
+        github = {
+            openLink(
+                context = context,
+                link = "https://github.com/HauntedMilkshake/fitness_app"
+            )
+        },
+        bugReport = {
+            openLink(
+                context = context,
+                link = "https://github.com/HauntedMilkshake/fitness_app/issues"
+            )
+        },
+        deleteAccount = { viewModel.deleteAccount() },
+        logout = { viewModel.logout() })
```

Comment:
trigger here navigateBack: () - No need to be ui state. But whenever you use it like you did Launched Effect is a must.

Reply:
moved it to on click

Reply:
Wouldn't that cause the ViewModel to be potentially garbage collected (and its jobs cancelled) before the login/delete account background work is done? 
While this may work most of the time I believe the way it was with the LaunchedEffect + having a loading indicator is safer for this and most use cases, especially in the delete account case where an error might need to be presented if the deletion fails.

Reply:
I agree - @Gotalicp add Launched effect in navigateBack block